[docs] Add extended search documentation (#64)
OpenCTI-Platform/docs: docs/usage/search.md

![Advanced search](assets/advanced-search.png)

> ### Extended search to uploaded files content

Reviewer: ```suggestion
### Extended search to upload files content
```

Author: uploaded files => files that are uploaded

!!! tip "Enterprise edition"

    Extended search to files content is available under the "Filigran entreprise edition" license.

    [Please read the dedicated page to have all information](../administration/enterprise.md)

> It's possible to extend the global search by keywords to the content of documents uploaded to the platform via the Data import tab, or directly linked to an entity via its Data tab.

Reviewer: ```suggestion
It's possible to extend the global search by keywords to the content of documents uploaded in the platform via the Data import tab, or directly linked to an entity via its Data tab.
```

Author: Generally it is `uploaded to` but we can use `uploaded on` 
e.g:
That video was `uploaded on` the internet
That video was `uploaded to` the website
...🤔

![Extended search](assets/extended-search-open.png)

#### Extended search requirements

**Files indexation requires to have one of these database configurations :**

* Elasticsearch >= 8.4
* Elasticsearch < 8.4 with [ingest-attachment plugin](https://www.elastic.co/guide/en/elasticsearch/plugins/8.3/ingest-attachment.html)
* OpenSearch with [ingest-attachment plugin](https://opensearch.org/docs/2.9/install-and-configure/plugins/)

Indexation of files is enabled by default and can be disabled in configuration file (`file_index_manager`).
It runs every 5 minutes to index last uploaded files.

Currently supported mime types : `text/plain`, `text/csv`, `application/pdf`, `application/vnd.ms-excel`, `application/vnd.openxmlformats-officedocument.spreadsheetml.sheet` (excel sheets).

## Bulk search

> The bulk search capabilities in available in the top bar of the platform and allow you to copy paste a list of keyword or objects (ie. list of domains, list of IP addresses, list of vulnerabilities, etc.) to search in the platform:

Reviewer: ```suggestion
The bulk search capabilities is available in the top bar of the platform and allows you to copy paste a list of keywords or objects (ie. list of domains, list of IP addresses, list of vulnerabilities, etc.) to search in the platform:
```
